### DISCUSSION TITLE
truncate version number in durexforth.prg

### DISCUSSION THREAD

**Discussion by jkotlinski** [jkotlinski/durexforth#586] [Ideas]
Hallo!

Since https://github.com/jkotlinski/durexforth/pull/345 , the version number is baked into durexForth, and displayed as a startup message.

I think it is nice to show the version at startup, but it is somewhat impractical, because the length of the version string changes depending on git status. This means the .prg size may change, and addresses jump around in RAM, for changes that are not code related. This is not so nice when debugging.

How would you feel about changing the version string, so that it always follows the format vN.N.N ? (i.e. skip the last digit, and -M). Then the version string will be a lot more stable. The "full" info could still be printed in disk listing.

**Comment by ekipan** [comment]
Another option:

```forth
\ file "version.fs"
: version s" v5.1.0-whatever" ;
```
```forth
\ towards the end of "base.fs"
include version
:noname version type cr ; start !
```

**Comment by ekipan** [comment]
> for changes that are not code related. This is not so nice when debugging.

I suppose the option above doesn't solve this for everything defined after `version`, just before it.

Hm. Besides truncating, you might also pad to a certain maximum size.

**Comment by jkotlinski** [comment]
I realized also without the tech perspective, I think it looks nicer with
plain vN.N.N in the startup message.

sön 22 dec. 2024 kl. 18:27 skrev ekipan ***@***.***>:

**Comment by cminter** [comment]
@jkotlinski , 

In the startup, you should go with what looks nice and the benefit to debug address stability.
